oarepo_requests/ui/theme/assets/semantic-ui/js/oarepo_requests_ui/request-detail/components/ConfirmModal.jsx

```diff
@@ -0,0 +1,117 @@
+
+  const onClose = () => {
+    setModalOpen(false);
+    setError(null);
+  };
+
+  const onSubmit = async (values, { setSubmitting }) => {
+    setError(null);
+    try {
+      await handleSubmit(values);
+      location.reload();
+    } catch (error) {
+      setError(error);
+    } finally {
+      setSubmitting(false);
+    }
+  };
+
+  return (
+    <Modal
+      className="requests-request-modal"
+      as={Dimmer.Dimmable}
+      blurring
+      onClose={onClose}
+      onOpen={() => setModalOpen(true)}
+      open={modalOpen}
+      trigger={triggerButton || <Button content="Open Modal" />}
+      closeOnDocumentClick={false}
+      closeOnDimmerClick={false}
+      role="dialog"
+      aria-labelledby="request-modal-header"
+    >
+      <Formik 
+        initialValues={{ 
+          payload: { 
+            content: "",
+            format: "html"
+          }
+        }}
+        onSubmit={onSubmit}
+      >
+        {({ isSubmitting, values, setFieldValue, setFieldTouched }) => (
+          <>
+            <Dimmer active={isSubmitting}>
+              <Loader inverted size="large" />
+            </Dimmer>
+            <Modal.Header as="h2" id="request-modal-header">{requestModalHeader ?? request?.type}</Modal.Header>
+            <Modal.Content>
+              {error && (
+                <Message error>
+                  <Message.Header>{i18next.t("Error while submitting comment.")}</Message.Header>
+                  <p>{error?.message}</p>
+                </Message>
+              )}
+              <Form id="submit-request-form">
+                <FormField>
+                  <RichInputField
+                    fieldPath="payload.content"
+                    label={
+                      <FieldLabel htmlFor="payload.content" label={`${i18next.t("Add comment")} (${i18next.t("optional")})`} className="rel-mb-25" />
+                    }
+                    optimized="true"
+                    placeholder={i18next.t('Your comment here...')}
+                    editor={
+                      <RichEditor
+                        value={values.payload.content}
+                        optimized
+                        onBlur={(event, editor) => {
+                          const cleanedContent = sanitizeInput(editor.getContent());
+                          setFieldValue("payload.content", cleanedContent);
+                          setFieldTouched("payload.content", true);
+                        }}
+                      />
+                    }
+                  />
+                </FormField>
+              </Form>
+            </Modal.Content>
+            <Modal.Actions>
+              <Button onClick={onClose} icon compact labelPosition="left" floated="left">
+                <Icon name="cancel" />
+                {i18next.t("Cancel")}
+              </Button>
+              {submitButton ?? React.cloneElement(triggerButton, { type: "submit", form: "submit-request-form" })}
```

Comment:
This looks like a hack :) also even triggerButton can be optional, passed as prop...this clones and tries to modify something this component knows nothing about, which feels like a bad practice. It assumes users of this component always pass in something that will respect what this cloning expects from it

Reply:
I would just render the submit button if available from prop, or render some sensible default

Reply:
Fixed in #34 with submitButton always being rendered if present (always present in this case).